packages/components/text/src/Text.tsx

```diff
@@ -123,6 +124,15 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
       fontSize: tokenStyle.fontSize * scaleFactor,
       lineHeight: tokenStyle.lineHeight * scaleFactor,
     };
+    if (scaleStyleAdjustments.fontSize > maximumFontSize) {
+      scaleStyleAdjustments = {
+        fontSize: maximumFontSize,
+        // Scale line height accordingly as well
+        // We use "as number" below because TypeScript's type checker thinks scaleStyleAdjustments.fontSize
+        // could be a named font size, even though we just wrote it as a number a few lines up.
+        lineHeight: scaleStyleAdjustments.lineHeight * (maximumFontSize / (scaleStyleAdjustments.fontSize as number)),
```

Comment:
Just wanted to note, I guess we're making an assumption here that the scale factor should be the same for the maximum font size?

Reply:
Yes. The maximum font size is specified in "real" points, so it's not affected by the scale factor.

(It's actually the other way around; the scale factor is effectively capped at `maximumFontSize / tokenStyle.fontSize`. Maybe it would be clearer to write it this way?)

Reply:
Just needed to think about this a bit more and makes more sense now - but if you think there's a clearer way to write it go for it